src/svm/interface/svm.c

```diff
@@ -1511,6 +1511,80 @@ PetscErrorCode SVMViewScore(SVM svm,PetscViewer v)
   PetscFunctionReturn(0);
 }
 
+#undef __FUNCT__
+#define __FUNCT__ "SVMSetMatGramian"
+/*@
+  SVMSetMatGramian - Set precomputed Gramian (kernel) matrix.
+
+  Input Parameters:
++ svm - SVM context
+- G - precomputed Gramian matrix
+
+  Level: intermediate
+
+.seealso SVM, SVMGetMatGramian(), SVMLoadMatGramian()
+@*/
+PetscErrorCode SVMSetMatGramian(SVM svm,Mat G)
+{
+
+  PetscFunctionBegin;
+  PetscValidHeaderSpecific(svm,SVM_CLASSID,1);
+  PetscValidHeaderSpecific(G,MAT_CLASSID,2);
+
+  TRY( PetscTryMethod(svm,"SVMSetMatGramian_C",(SVM,Mat),(svm,G)) );
+  PetscFunctionReturn(0);
+}
+
+#undef __FUNCT__
+#define __FUNCT__ "SVMGetMatGramian"
+/*@
+  SVMGetMatGramian - Get pracomputed Gramian (kernel) matrix.
+
+  Input Parameter:
+. svm - SVM context
+
+  Output Parameter:
+. G - precomputed Gramian matrix
+
+  Level: intermediate
+
+.seealso SVM, SVMSetMatGramian()
+@*/
+PetscErrorCode SVMGetMatGramian(SVM svm,Mat *G)
+{
+
+  PetscFunctionBegin;
+  PetscValidHeaderSpecific(svm,SVM_CLASSID,1);
+  PetscValidPointer(G,2);
+
+  TRY( PetscTryMethod(svm,"SVMGetMatGramian_C",(SVM,Mat *),(svm,G)) );
```

Comment:
Use vs Try. Does the user always expect to get G?
https://lists.mcs.anl.gov/pipermail/petsc-dev/2016-April/018920.html

Reply:
Okay, my fault. When the G is not available, it does not make sense to expect it. Fixed.